### PR TITLE
implement more application endpoints

### LINKED ISSUE #14: Methods for creating accounts and applications
A common use case for the support team is batch importing of user accounts and API keys (i.e. applications) and further updates.
For being able to use the client in Ruby scripts these methods for the following endpoints are required:
- Signup Express
- Application Create

Other endpoints necessary for further manipulation of the imported data:
- Application List
- Application Read
- Application Update

### PR DESCRIPTION
**Pull request by mikz**
closes #14

/cc @mayorova 

would be cool if you could try it before merging.

### CODE REVIEW

File: lib/3scale/api/client.rb [hunk 1]
```diff
@@ -36,6 +36,61 @@ def list_applications(service_id: nil)
...
+      # @param [String] email User Email
+      # @param [String] password User Password
+      # @param [Hash] attributes User and Account Attributes
+      # @option attributes [String] :account_plan_id Account Plan ID
+      # @option attributes [String] :service_plan_id Service Plan ID
+      # @option attributes [String] :application_plan_id Application Plan ID
+      # @option attributes [String] :username User Name. Defaults to User Email.
+      def signup(attributes = {}, name: , email: , password: , **rest)
+        body = { org_name: name,
+                 username: email,
```
Comment (mayorova) [inline lib/3scale/api/client.rb]: Why does username default to email here? I believe in the API spec it's a required field.

File: lib/3scale/api/client.rb [hunk 2]
```diff
@@ -36,6 +36,61 @@ def list_applications(service_id: nil)
...
+      # @api public
+      # @return [Hash] an Application
+      # @param [Fixnum] plan_id Application Plan ID
+      # @param [String] name Application Name
+      # @param [String] description Application Description
+      # @param [Hash] attributes Application Attributes
+      # @option attributes [String] :user_key Application User Key
+      # @option attributes [String] :application_id Application App ID
+      # @option attributes [String] :application_key Application App Key(s)
+      # @option attributes [String] :username User Name. Defaults to User Email.
```
Comment (mayorova) [inline lib/3scale/api/client.rb]: Doesn't belong here :)

**Comment by mayorova** [comment]
@mikz Tried through `bin/console`, and it works properly.
Thanks!

File: lib/3scale/api/client.rb (hunk 1 above)
Comment (mikz) [inline lib/3scale/api/client.rb]: It is required by the API, yes. I felt having less parameters would be nicer api. You can still pass it:

```ruby
signup(username: 'foobar', name: 'name', email: 'foo@example.com')
```

But if you feel like that is hiding something, I'll make it explicit. I feel like giving user the username is a step backwards and we actually should not have it as required field and just default to the email on the API level.

File: lib/3scale/api/client.rb (hunk 2 above)
Comment (mikz) [inline lib/3scale/api/client.rb]: true, good catch!

File: lib/3scale/api/client.rb (hunk 1 above)
Comment (mayorova) [inline lib/3scale/api/client.rb]: Yes, maybe it shouldn't be a required required in the API. But as long as it is, I think the client should also do it, it's more intuitive. Well, it's my personal preference, not sure about the best practices.

Comment (MarkCheshire) [inline lib/3scale/api/client.rb]: Email is actually optional and not required

Comment (mikz) [inline lib/3scale/api/client.rb]: Correct. From my tests only name & username are required. Password is also not required.

**Comment by mikz** [comment]
@mayorova changes applied, thanks for review! will merge & release very soon

**Comment by mikz** [comment]
This has been released as `0.1.4`.

**Pull request merged**